src/diffusers/pipelines/alt_diffusion/pipeline_alt_diffusion.py

```diff
@@ -417,17 +420,26 @@ def _encode_prompt(
 
         return prompt_embeds
 
-    def run_safety_checker(self, image, device, dtype):
-        if self.safety_checker is not None:
-            safety_checker_input = self.feature_extractor(self.numpy_to_pil(image), return_tensors="pt").to(device)
+    def run_safety_checker(self, image, device, dtype, output_type="pil"):
+        if self.safety_checker is None or output_type == "latent":
+            has_nsfw_concept = False
+        else:
+            image = self.image_processor.postprocess(image, output_type="pt")
+            feature_extractor_input = self.image_processor.postprocess(image, output_type="pil")
```

Comment:
Wouldn't this denormalize twice?

Reply:
the `if image.min() < 0:` condition make sure it only denormalize once

Reply:
I think here is a misunderstanding. The `postprocess` function should **always** apply the normalization `x / 2 + 0.5` - We should not have to run this function twice here

Reply:
@pcuenca can you take a look here?
my main confusion is that in order to carry out the refactor Patrick suggested in https://github.com/huggingface/diffusers/pull/2943#issuecomment-1495860454 (i.e. move the `x / 2 + 0.5` to `postprocess` function), I think the `postprocess` function will have to check if the image is already denormalized and not denormalize twice if it's the case (explained here https://github.com/huggingface/diffusers/pull/2943#discussion_r1163241978)

Did I understand something wrong? is there a better way to check if the image is already denormalized?

Reply:
I think it's just that this:
`"is there a better way to check if the image is already denormalized"` 
that needs to be addressed, we cannot check here depending on the values of the tensor whether it has already been denormalized.

Imagine for some reason stable diffusion generates a tensor that is always larger 0 - in this case we would never denormalize which is bad. 

A better check is to add a flag argument to the post-process function that allows the safety checker to not denormalize it

Reply:
Yes, I agree with Patrick, I'd rather not chain calls to `postprocess`. I see two potential solutions:
* Just convert the post-processed image to PIL (maybe we can add a helper to the postprocessor if we need to).
* Maybe just swap the calls? Perhaps I'm confused but wouldn't this work?

```Python
feature_extractor_input = self.image_processor.postprocess(image, output_type="pil")
image = self.image_processor.postprocess(image, output_type="pt")
```

Reply:
@pcuenca yes we can swap the calls here. I agree it's better to swap and it will avoid calling `postprocess` twice inside `run_safety_checker`. 

I still need to call `postprocess` again with the output of `run_safety_checker`, which is already denormalized https://github.com/huggingface/diffusers/blob/postprocessing-refactor/src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py#L712

Reply:
@patrickvonplaten I think the flag will solve my problem - I will go with that for now. We can discuss later and happy to refactor again